homeassistant/components/camera/canary.py

```diff
@@ -25,71 +36,82 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     devices = []
 
     for location in data.locations:
-        entries = data.get_motion_entries(location.location_id)
-        if entries:
-            devices.append(CanaryCamera(data, location.location_id,
-                                        DEFAULT_TIMEOUT))
+        for device in location.devices:
+            if device.is_online:
+                devices.append(
+                    CanaryCamera(hass, config, data, location, device,
+                                 DEFAULT_TIMEOUT))
 
     add_devices(devices, True)
 
 
 class CanaryCamera(Camera):
     """An implementation of a Canary security camera."""
 
-    def __init__(self, data, location_id, timeout):
+    def __init__(self, hass, config, data, location, device, timeout):
         """Initialize a Canary security camera."""
         super().__init__()
+
+        self._ffmpeg = hass.data[DATA_FFMPEG]
+        self._ffmpeg_arguments = config.get(CONF_FFMPEG_ARGUMENTS)
         self._data = data
-        self._location_id = location_id
+        self._location = location
+        self._device = device
         self._timeout = timeout
-
-        self._location = None
-        self._motion_entry = None
-        self._image_content = None
-
-    def camera_image(self):
-        """Update the status of the camera and return bytes of camera image."""
-        self.update()
-        return self._image_content
+        self._live_stream_session = None
 
     @property
     def name(self):
         """Return the name of this device."""
-        return self._location.name
+        return self._device.name
 
     @property
     def is_recording(self):
         """Return true if the device is recording."""
         return self._location.is_recording
 
-    @property
-    def device_state_attributes(self):
-        """Return device specific state attributes."""
-        if self._motion_entry is None:
-            return None
-
-        return {
-            ATTR_MOTION_START_TIME: self._motion_entry.start_time,
-            ATTR_MOTION_END_TIME: self._motion_entry.end_time,
-        }
-
-    def update(self):
-        """Update the status of the camera."""
-        self._data.update()
-        self._location = self._data.get_location(self._location_id)
-
-        entries = self._data.get_motion_entries(self._location_id)
-        if entries:
-            current = entries[0]
-            previous = self._motion_entry
-
-            if previous is None or previous.entry_id != current.entry_id:
-                self._motion_entry = current
-                self._image_content = requests.get(
-                    current.thumbnails[0].image_url,
-                    timeout=self._timeout).content
-
     @property
     def motion_detection_enabled(self):
         """Return the camera motion detection status."""
         return not self._location.is_recording
+
+    @property
+    def should_poll(self):
+        """Update the image periodically."""
+        return True
```

Comment:
Why do you need polling?

Reply:
This is so that live stream session can be extended periodically

Reply:
Set a throttle on the method that does that and just call it before getting the image. Ie, don't activate poll here. The frontend should take care of this.